compiler-access/src/main/scala/org/alephium/ralph/lsp/access/util/StringUtil.scala

```diff
@@ -29,4 +32,7 @@ object StringUtil {
   def codeLines(code: String): Array[String] =
     code.split("\r\n|\r|\n")
 
+  def lineSeparatorLength(code: String): Int =
+    if(code.contains("\r\n")) 2 else 1
```

Comment:
Hey @tdroxler, sorry I forgot to ask before. The function call `contains`, is it necessary? We can't check if the OS is Windows and use the length `2` and `1` otherwise? The `computeIndex` function which calls this function is called on every keystroke, if we can reduce the amount of work needed here, the better.

Reply:
Unfortunately that's the problem I wrote in the description, from my testing on windows, I was actually receiving code with `\n` only 
![image](https://github.com/alephium/ralph-lsp/assets/2979182/2c8640a5-b3a5-4bab-b4d1-04f1d760a5e0)
So Windows doesn't always use 2 :scream: I guess it depends on the client.

I will check if there's a config available in the `initialize` request, maybe there we can announce which separator we use?